src/components/TodosForm/TodosForm.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Dispatch,
+  SetStateAction,
+  ChangeEvent,
+  FormEvent,
+  useState,
+} from 'react';
+
+import users from '../../api/users';
+import { getNextTodoId } from '../../utils/todo/getNextTodoId';
+import { ITodo } from '../../types/todo';
+
+interface IProps {
+  todos: ITodo[];
+  setTodos: Dispatch<SetStateAction<ITodo[]>>;
+}
+
+const TodosForm = ({ todos, setTodos }: IProps) => {
+  const [title, setTitle] = useState<string | null>('');
+  const [userId, setUserId] = useState<number | null>(0);
+
+  const handleReset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const checkTodoFormState = () => {
+    if (!title) {
+      setTitle(null);
+    }
+    if (!userId) {
+      setUserId(null);
+    }
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title || !userId) {
+      return checkTodoFormState();
+    }
+
+    const todo = {
+      id: getNextTodoId(todos),
+      title,
+      userId,
+      completed: false,
+    };
+
+    setTodos((prevState: ITodo[]) => [...prevState, todo]);
+    handleReset();
+  };
+
+  const handleChangeTitle = (e: ChangeEvent<HTMLInputElement>) =>
+    setTitle(e.target.value);
+
+  const handleChangeUserId = (e: ChangeEvent<HTMLSelectElement>) =>
+    setUserId(+e.target.value);
+
+  return (
+    <form action="/api/todos.ts" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <input
+          type="text"
+          data-cy="titleInput"
+          name="title"
+          value={title !== null ? title : ''}
+          onChange={handleChangeTitle}
+          placeholder="Enter the title"
+        />
+
+        {title === null && <span className="error">Please enter a title</span>}
```

Comment:
Okay, I think I get it. If you do it to handle errors, you can just add a state for the errors and depending on it show an error message. Because it looks strange